src/components/Quotes.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+// import PropTypes from 'prop-types';
+import RenderQuotes from './childComponents/RenderQuotes';
+
+function Quotes() {
+  const [quotes, setQuotes] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const names = ['inspirational', 'intelligence', 'knowledge'];
+    const randomCategory = names[Math.floor(Math.random() * names.length)];
+
+    const apiKey = 'RN8eEX+Te+lyH3p9oof/ZA==wvp6dHPhWpLW8yd2';
+    fetch(`https://api.api-ninjas.com/v1/quotes?category=${randomCategory}`, {
+      headers: { 'X-Api-Key': `${apiKey}` },
+    })
+      .then((resp) => resp.json())
+      .then((json) => {
+        setQuotes(json);
+        setLoading(false);
+      })
+      .catch((error) => {
+        setError(error);
+        setLoading(false);
+      });
+  }, []);
+
```

Comment:
- [ ] Great work so far! 👏👏 However, when an incorrect API key is used, instead of receiving an error message, the following error occurs ⚠. Please take a look at this issue and handle the error state correctly to ensure that the appropriate message is displayed, enhancing the overall user experience

![image](https://github.com/diarisdiakite/DD-math-magicians/assets/106119105/27786529-6451-4f62-93d4-90b0245fda79)

You can consider using something like below to to check if the response was successful or not

```ruby
  useEffect(() => {
    const names = ['inspirational', 'intelligence', 'knowledge'];
    const randomCategory = names[Math.floor(Math.random() * names.length)];

    const apiKey = 'RN8eEX+Te+lyH3p9oof/ZA==wvp6dHPhWpLW8yd';
    fetch(`https://api.api-ninjas.com/v1/quotes?category=${randomCategory}`, {
      headers: { 'X-Api-Key': apiKey },
    })
      .then((resp) => {
        if (!resp.ok) {
          throw new Error('Failed to fetch quotes');
        }
        return resp.json();
      })
      .then((json) => {
        setQuotes(json);
        setLoading(false);
      })
      .catch((error) => {
        setError(error);
        setLoading(false);
      });
  }, []);
```

_Please make sure to test this before asking for another review_